src/screens/login/components/functions/clouderyBackgroundFetcher.ts

```diff
@@ -0,0 +1,34 @@
+import type { WebViewMessageEvent } from 'react-native-webview'
+
+interface ClouderyMessage {
+  message: string
+  color?: string
+}
+
+type SetBackgroundColorCallback = (color: string) => void
+
+export const fetchBackgroundOnLoad = `
+(function() {
+  window.addEventListener("load", function(event) {
+    const color = getComputedStyle(
+      document.getElementsByTagName("main")[0]
+    ).getPropertyValue('--paperBackgroundColor')
+
+    window.ReactNativeWebView.postMessage(JSON.stringify({
+      message: 'setBackgroundColor',
+      color: color
+    }))
+  })
+})();
+`
+
+export const tryProcessClouderyBackgroundMessage = (
+  event: WebViewMessageEvent,
+  setBackgroundColor: SetBackgroundColorCallback
+): void => {
+  const message = JSON.parse(event.nativeEvent.data) as ClouderyMessage
+
+  if (message.message === 'setBackgroundColor' && message.color !== undefined) {
+    setBackgroundColor(message.color)
+  }
+}
```

Comment:
I really like this approach, but what are the performances of it? Waiting for "load" and the "getComputedStyle" that request a reflow, I assume it's pretty slow to get the information. Don't we have visually "glitch"?

Reply:
For now I didn't see anyglitch but I didn't test every scenario (also i tested only iOS).

The base scenario should not be problematic because those view are loaded behind the Welcome screen (which has its own colored bars).

But the color may change in foreground when we will implement the "postcode choice" scenario. I tried to simulate it and it seems ok on iOS. But I would prefer to test it in real condition